server/src/main/java/org/opensearch/search/aggregations/support/MultiTermsValuesSourceConfig.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.aggregations.support;
+
+import org.opensearch.common.ParseField;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ObjectParser;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.script.Script;
+import org.opensearch.search.aggregations.AggregationBuilder;
+import org.opensearch.search.aggregations.bucket.terms.IncludeExclude;
+
+import java.io.IOException;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.Objects;
+
+/**
+ * A configuration that tells multi-terms aggregations how to retrieve data from index
+ * in order to run a specific aggregation.
+ *
+ * Similar with {@link MultiValuesSourceFieldConfig}, but support value script.
+ */
+public class MultiTermsValuesSourceConfig implements Writeable, ToXContentObject {
```

Comment:
Can we abstract this as well so that `MultiValuesSourceFieldConfig` and `MultiTermsValuesSourceFieldConfig` extend from a common `BaseMultiValuesSourceFieldConfig` w/o duplicating so much?

Reply:
Are we not supporting the ability to specify `filter` to further filter on the values like `MultiTermsValuesSourceFieldConfig`? Will that be a follow on enhancement?

Reply:
1. add BaseMultiValuesSourceFieldConfig to remove duplication.
2. [multi_terms aggregation use include/exclude to filter values](https://github.com/penghuo/OpenSearch/blob/multi-terms-agg/server/src/test/java/org/opensearch/search/aggregations/bucket/terms/MultiTermsAggregatorTests.java#L831). [similar to include/exclude terms aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/search-aggregations-bucket-terms-aggregation.html#_filtering_values_4).  I found filter is disabled in [WeightedAvgAggregationBuilder](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/aggregations/metrics/WeightedAvgAggregationBuilder.java#L72) and I didn't found filter been used in any other aggregation.